lib/ansible/modules/cloud/amazon/iam_password_policy.py

```diff
@@ -0,0 +1,189 @@
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+
+class IAMConnection(object):
+
+    def __init__(self, module, region, **aws_connect_params):
+        try:
+            self.connection = boto3_conn(module, conn_type='resource',
+                                         resource='iam', region=region,
+                                         **aws_connect_params)
+            self.module = module
+        except Exception as e:
+            module.fail_json(msg="Failed to connect to AWS: %s" % str(e))
+        self.region = region
+
+    def update_password_policy(self, module, policy):
+        min_pw_length = module.params.get('min_pw_length')
+        require_symbols = module.params.get('require_symbols')
+        require_numbers = module.params.get('require_numbers')
+        require_uppercase = module.params.get('require_uppercase')
+        require_lowercase = module.params.get('require_lowercase')
+        allow_pw_change = module.params.get('allow_pw_change')
+        pw_max_age = module.params.get('pw_max_age')
+        pw_reuse_prevent = module.params.get('pw_reuse_prevent')
+        pw_expire = module.params.get('pw_expire')
+
+        try:
+            results = policy.update(
+                MinimumPasswordLength=min_pw_length,
+                RequireSymbols=require_symbols,
+                RequireNumbers=require_numbers,
+                RequireUppercaseCharacters=require_uppercase,
+                RequireLowercaseCharacters=require_lowercase,
+                AllowUsersToChangePassword=allow_pw_change,
+                MaxPasswordAge=pw_max_age,
+                PasswordReusePrevention=pw_reuse_prevent,
+                HardExpiry=pw_expire
+            )
+            policy.reload()
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self.module.fail_json_aws(e, msg="Couldn't update IAM Password Policy")
+        return camel_dict_to_snake_dict(results)
+
+    def delete_password_policy(self, policy):
+        try:
+            results = policy.delete()
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                self.module.exit_json(changed=False, task_status={'IAM': "Couldn't find IAM Password Policy"})
+            else:
+                self.module.fail_json_aws(e, msg="Couldn't delete IAM Password Policy")
+        return camel_dict_to_snake_dict(results)
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'state': dict(choices=['present', 'absent'], required=True),
+            'min_pw_length': dict(type='int', default=6),
+            'require_symbols': dict(type='bool', default=False),
+            'require_numbers': dict(type='bool', default=False),
```

Comment:
Because these have hard defaults (`False` rather than `None`) this module would not always be idempotent if all options aren't provided. I think it should only alter options that the user passes to the module, rather than resetting to defaults.